File: performance-benchmark/src/test/scala/net/ssimmie/todos/performance/benchmark/HealthCheckAction.scala
package net.ssimmie.todos.performance.benchmark

import io.gatling.core.Predef._
import io.gatling.core.structure.ChainBuilder
import io.gatling.http.HeaderNames._
import io.gatling.http.HeaderValues.ApplicationJson
import io.gatling.http.Predef._

object HealthCheckAction {

  val healthcheck: ChainBuilder = exec(http("Healthcheck application")
    .get("/actuator/health")
    .header(ContentType, ApplicationJson)
    .header(Accept, ApplicationJson)
    .check(status.is(200))
    .check(jsonPath("$.status").is("UP")))

}

File: performance-benchmark/src/test/java/net/ssimmie/todos/performance/benchmark/HealthCheckAction.java
package net.ssimmie.todos.performance.benchmark;

import io.gatling.javaapi.core.ChainBuilder;

import static io.gatling.javaapi.core.CoreDsl.*;
import static io.gatling.javaapi.http.HttpDsl.*;

public class HealthCheckAction {

    public static ChainBuilder healthcheck() {
        return exec(http("Healthcheck application")
                .get("/actuator/health")
                .header("Content-Type", "application/json")
                .header("Accept", "application/json")
                .check(status().is(200))
                .check(jsonPath("$.status").is("UP"))
        );
    }
}